### PR TITLE
Implemented tensorflow l2_normalize in the tensorflow frontend and couldnt test it .

### PR DESCRIPTION
**Pull request by ferstappen23**
**This pull request makes the following changes :**

* Implements issue #15112
**Problems** 
 I got an initial error message ImportError while loading conftest C:\Users\Verstappen\ivy\ivy_tests\test_ivy\conftest.py   , indicating that there is an error while loading the configuration file 'conftest.py' . However, the error further showed that the error occurred in the file function_testing.py on line 39, when attempting to import the tensorflow module, suggesting that there is an attribute error because of a circular import between the jax and tensorflow modules,which is causing the function_testing.py to fail .
Do you guys know how to resolve this?

### CODE REVIEW

**Comment by ferstappen23** [comment]
Hi @theRealBird,
I finished my task and did my PR , I also checked the logs and fixed the keyword error for the get_shape helper function.
Can you review it for me.
Thanks

**Comment by theRealBird** [comment]
Hi @ferstappen23, Thank you for your patience. I will definitely have a look at your PR. It seems you were having some problems before. I will have a look and update you.
Thanks!

**Comment by ferstappen23** [comment]
I would highly appreciate it .

File: ivy_tests/test_ivy/conftest.py
Comment (theRealBird) [inline ivy_tests/test_ivy/conftest.py:1]: Are these changes intended?😅

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
-
+import pytest
 # local
 import ivy_tests.test_ivy.helpers as helpers
 from ivy_tests.test_ivy.test_functional.test_core.test_statistical import (
     statistical_dtype_values,
 )
 from ivy_tests.test_ivy.helpers import handle_frontend_test
 
+import paddle.fluid.core
```
Comment (theRealBird) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:13]: How is ```paddle.fluid.core``` being used in the code?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 2]
```diff
@@ -2527,3 +2530,36 @@ def test_tensorflow_in_top_k(
...
+    *,
+    dtype_and_x,
+    shape_and_axis,
+    #eps,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device,
+):
+    import pdb
```
Comment (theRealBird) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:2557]: I think you were using a debugger here. This code should be removed after local testing is finished.

File: requirements/optional.txt
Comment (theRealBird) [inline requirements/optional.txt:1]: Are these changes related to the implementation of ```l2_normalize```?

File: .idea/vcs.xml
Comment (theRealBird) [inline .idea/vcs.xml:1]: Did you mean to commit the changes in these files? I think the .idea folder can be ignored.

File: ivy/functional/frontends/tensorflow/math.py
Comment (theRealBird) [inline ivy/functional/frontends/tensorflow/math.py:1]: ```tf.math.l2_normalize``` also supports the arguments ```epsilon``` and ```name``` as seen in their documentation [here](https://www.tensorflow.org/api_docs/python/tf/math/l2_normalize) 

Hence, the frontend implementation of this function should also support those requirements. Please add support for the same.

File: Dockerfile
Comment (theRealBird) [inline Dockerfile:1]: Could you please explain why these changes are needed? Thanks!

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 3]
```diff
@@ -2,14 +2,17 @@
 import ivy
 import numpy as np
 from hypothesis import strategies as st, assume
-
+import pytest
```
Comment (theRealBird) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:5]: Are we using pytest later in our code?

File: run_tests_CLI/run_tests.sh
Comment (theRealBird) [inline run_tests_CLI/run_tests.sh:1]: Could you please explain what these changes do? Thanks :)

**Review by theRealBird — COMMENTED** [review]
Hi @ferstappen23, 

I have left some comments on each of the files that are a part of this commit. I understand that you were facing some issues earlier.

Ideally, the implementation of a frontend function like ```l2_normalize``` would not require changes to so many files 😅

My initial thought is that changes to ```ivy/functional/frontends/tensorflow/math.py``` and ```ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py``` should be enough for the implementation.

I would suggest having a look at a similar implementation of ```l2_normalization``` (alias in tf.linalg) in ``` ivy/functional/frontends/tensorflow/linalg.py``` and its corresponding test if you feel stuck or checking out the setting-up section of our docs to ensure your tools are working correctly.

Thank you so much for your work and patience. Let me know if you need any help 😃

**Comment by ferstappen23** [comment]
@theRealBird 
 Thank you so much for your feedback , i agree that the changes only concerns the math.py and the test_math.py files ,the unintended changes were made in my attempts  to ensure the functionality of the tests. , i'm aware now that i should have staged and committed only the relevant files. I will promptly address this and make the necessary changes.
